.github/workflows/linux.yml

```diff
@@ -99,23 +99,40 @@ jobs:
   # Runs the underlying job (“workload”) on a self-hosted runner if available,
   # with the help of a `runner-select` job and a `runner-timeout` job.
   runner-select:
-    uses: ./.github/workflows/self-hosted-runner-select.yml
-    secrets: inherit
-    with:
-      # Before updating the GH action runner image for the nightly job, ensure
-      # that the system has a glibc version that is compatible with the one
-      # used by the wpt.fyi runners.
-      github-hosted-runner-label: ubuntu-22.04
-      self-hosted-image-name: servo-ubuntu2204
-      force-github-hosted-runner: ${{ inputs.upload || inputs.force-github-hosted-runner }}
+    outputs:
+      unique-id: ${{ steps.select.outputs.unique_id }}
+      selected-runner-label: ${{ steps.select.outputs.selected_runner_label }}
+      is-self-hosted: ${{ steps.select.outputs.is_self_hosted }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: delan/servo-ci@main
+      - name: Select and reserve best available runner
+        id: select
+        # Before updating the GH action runner image for the nightly job, ensure
+        # that the system has a glibc version that is compatible with the one
+        # used by the wpt.fyi runners.
+        run: |
+          ci runner select \
+            --github-repository '${{ github.repository }}' \
+            --github-run-id '${{ github.run_id }}' \
+            --monitor-api-token '${{ secrets.SERVO_CI_MONITOR_API_TOKEN }}' \
+            --github-hosted-runner-label 'ubuntu-22.04' \
+            --self-hosted-image-name 'servo-ubuntu2204' \
+            --force-github-hosted-runner '${{ inputs.upload || inputs.force-github-hosted-runner }}'
```

Comment:
I think first two args are provided via env too somehow (`GITHUB_CONTEXT` IIRC).

Also I think packing this into composite action would make this clearer.